llama-index-core/llama_index/core/workflow/session.py

```diff
@@ -0,0 +1,68 @@
+import asyncio
+from typing import Any, TYPE_CHECKING, Dict, Set, List, Tuple, Optional
+
+
+from .context import Context
+from .decorators import StepConfig
+from .events import Event
+from .errors import WorkflowRuntimeError
+
+if TYPE_CHECKING:
+    from .workflow import Workflow
+
+
+class WorkflowSession:
+    def __init__(self, workflow: "Workflow") -> None:
+        self._workflow = workflow
+        # Broker machinery
+        self._queues: Dict[str, asyncio.Queue] = {}
+        self._tasks: Set[asyncio.Task] = set()
+        self._broker_log: List[Event] = []
+        self._step_flags: Dict[str, asyncio.Event] = {}
+        self._accepted_events: List[Tuple[str, str]] = []
+        self._retval: Any = None
+        self._root_context = Context(self)
+        # Context management
+        self._step_to_context: Dict[str, Context] = {}
+
+    def send_event(self, message: Event, step: Optional[str] = None) -> None:
```

Comment:
so if step is not None, we are sending the event only to that step (assuming it accepts that event?)

Reply:
Yes, this is from https://github.com/run-llama/llama_index/pull/15259